src/main/java/EventsAndCommands/ChatEventDistributor.java

```diff
@@ -0,0 +1,74 @@
+package EventsAndCommands;
+
+import java.io.DataInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import EventsAndCommands.FunEvents.GoodBotEvent;
+import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+
+import org.reflections.*;
+
+/**
+ * CommandHandler
+ */
+public class ChatEventDistributor extends ListenerAdapter{
+
+    private static ChatEventDistributor instance = null;
+
+    private Set<ChatEventHandler> eventHandlers;
+
+    public static ChatEventDistributor getInstance(){
```

Comment:
This is not thread safe, but JDA is quite multi threaded. Not sure where it is used, but please add at least a doc comment.

And I am not sure a singleton is a good fit. If you pass an initialization consumer to the `load` method, you can make this a normal class. This works as you can create one instance in the main class and just register it in the initialization consumer. This also removes the dependency of the listener to the manager. If you want to keep it, you can inject the instance in the initialization block.

Reply:
Has ben resolved since the distribution of events to the fitting handlers is now handled by JDA itself again